### PR TITLE
docs: avoid using yo 5.x for plugin developer

### LINKED ISSUE #4616: generator-verdaccio-plugin cannot work at yo v5.x
**Your Environment**
  
 * **verdaccio version**: 5.30.3
 * **node version** [18.19.1]:
 * **package manager**: [npm@10, pnpm@9]
 * **os**: [windows@11]
 * **platform**: [npm]

**Describe the bug**

generate-verdaccio-plugin should only work in yo v4.x, not v5.x, but the document does not point this problem

**To Reproduce**

```shell
npm i -g yo

npm i -g generator-verdaccio-plugin

yo verdaccio-plugin
```

**Expected behavior**

Show as normal:

image

**Screenshots, server logs, package manager log**

```text
❯ pnpm start

> verdaccio-yo-demo@ start /home/projects/rollup-repro-rjuaav
> yo verdaccio-plugin

Error verdaccio-plugin 

Current environment doesn't provides some necessary feature this generator needs.
 ELIFECYCLE  Command failed with exit code 1.
```

**Configuration File (cat ~/.config/verdaccio/config.yaml)**

**Environment information**

Environment Info:
  System:
    OS: Windows 11 10.0.22631
    CPU: (16) x64 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
  Binaries:
    npm: 10.2.4 - ~\AppData\Local\pnpm\npm.CMD
    pnpm: 9.0.5 - ~\AppData\Local\pnpm\pnpm.EXE
  Virtualization:
    Docker: 25.0.3 - C:\Program Files\Docker\Docker\resources\bin\docker.EXE
  Browsers:
    Edge: Chromium (123.0.2420.97)

**Debugging output**

This information is not pertinent to Verdaccio details, so I won't include it here to avoid wasting your time.

**Contribute to Verdaccio**

- [x] I'm willing to fix this bug 🥇

### PR DESCRIPTION
**Pull request by gweesin**
closes #4616

### CODE REVIEW

File: website/docs/plugin-generator.md [hunk 1]
```diff
@@ -19,6 +19,12 @@ then, install the custom generator running the following in your terminal.
 npm i -g generator-verdaccio-plugin
 ```
 
+:::caution
+
+generator-verdaccio-plugin does not compality yeoman version 5 now, you must use version 4 correctly.
```
Comment (juanpicado) [inline website/docs/plugin-generator.md:24]: What exactly fails ?

**Review by juanpicado — COMMENTED** [review]
If you want to show this on the latest you might need also to update the 5.x / 6.x version pages. Otherwise will be displayed only on next section.

File: website/docs/plugin-generator.md (hunk 1 above)
Comment (gweesin) [reply website/docs/plugin-generator.md:24]: I've been debugging both the generator-verdaccio-plugin@4.1.0 and the latest version of generator-verdaccio-plugin locally. The issue I've encountered is that yeoman-api isn't compatible with yeoman-generator@5.7.0.

local yeoman-generate version:
image

generator-verdaccio-plugin@4.1.0 yeoman-generate version:
image

local version is fixed by verdaccio/generator-verdaccio-plugin#186
So, all you need to do is to publish the 'generator-verdaccio-plugin' based on the current content to fix this issue.🤣

Comment (juanpicado) [reply website/docs/plugin-generator.md:24]: I will :) maybe I forgot doing it

Comment (gweesin) [reply website/docs/plugin-generator.md:24]: Yes, you can close this pr when the new plugin generator released.
Have a good time on your holiday!

**Review by mbtools — APPROVED** [review]
LGTM 👍

**Pull request merged**